File: GalileoX_CAAD002X_Android_App_Development_for_Beginners/MicroMaster_ContactList-BaseCodeExercise_Lesson1/app/src/main/java/com/beginner/micromaster/contactlist/DetailActivity.java
package com.beginner.micromaster.contactlist;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import java.util.Locale;

public class DetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        TextView contactName = (TextView) findViewById(R.id.contact_name);
        TextView contactLastName = (TextView) findViewById(R.id.contact_last_name);
        TextView contactEmail = (TextView) findViewById(R.id.contact_email);
        TextView contactPhone = (TextView) findViewById(R.id.contact_phone);

        Contact contact = getIntent().getExtras().getParcelable(Contact.class.getSimpleName());

        contactName.setText(contact.getName());
        contactLastName.setText(contact.getLastName());
        contactEmail.setText(contact.getEmail());
        contactPhone.setText(String.format(Locale.getDefault(), "%d", contact.getPhoneNumber()));
    }
}

File: WeatherApp/app/src/main/java/com/example/robertduriancik/weatherapp/ui/activities/DetailActivity.kt
package com.example.robertduriancik.weatherapp.ui.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import com.example.robertduriancik.weatherapp.R
import com.example.robertduriancik.weatherapp.domain.commands.RequestDayForecastCommand
import com.example.robertduriancik.weatherapp.domain.model.Forecast
import com.example.robertduriancik.weatherapp.extensions.color
import com.example.robertduriancik.weatherapp.extensions.textColor
import com.example.robertduriancik.weatherapp.extensions.toDateString
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_detail.*
import org.jetbrains.anko.ctx
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread
import java.text.DateFormat

class DetailActivity : AppCompatActivity() {

    companion object {
        const val ID = "DetailActivity:id"
        const val CITY_NAME = "DetailActivity:cityName"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        title = intent.getStringExtra(CITY_NAME)

        doAsync {
            val result = RequestDayForecastCommand(intent.getLongExtra(ID, -1)).execute()
            uiThread { bindForecast(result) }
        }
    }

    private fun bindForecast(forecast: Forecast) = with(forecast) {
        Picasso.with(ctx).load(iconUrl).into(icon)
        supportActionBar?.subtitle = date.toDateString(DateFormat.FULL)
        weatherDescription.text = description
        bindWeather(high to maxTemperature, low to minTemperature)
    }

    private fun bindWeather(vararg views: Pair<Int, TextView>) = views.forEach {
        it.second.text = "${it.first}"
        it.second.textColor = color(when (it.first) {
            in -50..0 -> android.R.color.holo_red_dark
            in 0..15 -> android.R.color.holo_orange_dark
            else -> android.R.color.holo_green_dark
        })
    }
}
